apps/dashboard/src/components/activity/activity-logs.tsx

```diff
@@ -1,34 +1,225 @@
+  const { currentEnvironment } = useEnvironment();
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+
+  const resentMetadata = {
+    __resent_transaction_id: activity.transactionId,
+    __resent_at: new Date().toISOString(),
+  };
+  const resentPayload = payload ? { ...payload, ...resentMetadata } : resentMetadata;
+  const formattedPayload = payload ? JSON.stringify(payload, null, 2) : '{}';
+
+  const setIsFullscreenOpen = useCallback((isOpen: boolean) => {
+    if (isOpen && popoverCloseRef.current) {
+      popoverCloseRef.current.click();
+    }
+
+    setIsFullscreenOpenState(isOpen);
+  }, []);
+
+  const closePopover = useCallback(() => {
+    if (popoverCloseRef.current) {
+      popoverCloseRef.current.click();
+    }
+
+    setIsPopoverOpen(false);
+  }, []);
+
+  const handleResend = useCallback(async () => {
+    try {
+      setIsResending(true);
+
+      const {
+        data: { transactionId: newTransactionId },
+      } = await triggerWorkflow({
+        name: activity.template?.triggers[0].identifier ?? '',
+        to: activity.subscriber?.subscriberId,
+        payload: resentPayload,
+        environment: { _id: activity._environmentId } as IEnvironment,
+      });
+
+      setIsResending(false);
+
+      if (!newTransactionId) {
+        return showToast({
+          variant: 'lg',
+          children: ({ close }) => (
+            <>
+              <ToastIcon variant="error" />
+              <div className="flex flex-col gap-2">
+                <span className="font-medium">Test workflow failed</span>
+                <span className="text-foreground-600 inline">
+                  Workflow <span className="font-bold">{activity.template?.name}</span> cannot be triggered. Ensure that
+                  it is active and requires not further actions.
+                </span>
+              </div>
+              <ToastClose onClick={close} />
+            </>
+          ),
+          options: {
+            position: 'bottom-right',
+          },
+        });
+      }
+
+      closePopover();
+      setIsFullscreenOpen(false);
+
+      toast.success('Notification resent successfully', {
+        description: `A new notification has been triggered with transaction ID: ${newTransactionId}`,
+      });
+
+      const checkAndUpdateTransaction = async () => {
+        if (currentEnvironment) {
+          const { data: activities } = await getActivityList({
+            environment: currentEnvironment,
+            page: 0,
+            limit: 1,
+            filters: {
+              transactionId: newTransactionId,
+            },
+          });
+
+          if (activities.length > 0) {
+            queryClient.invalidateQueries({
+              queryKey: [QueryKeys.fetchActivities, activity._environmentId],
+            });
+            onTransactionIdChange?.(newTransactionId, activities[0]._id);
+            return;
+          }
+        }
+      };
+
+      setTimeout(checkAndUpdateTransaction, 1000);
```

Comment:
this one is a bit hacky, what if I have the other activities triggered at the same time? it might pick a wrong one

Reply:
agree, but i reused what we have today. today we use useFetchActivities with refetchInterval, in this scenario, we need to wait a bit before we fetch activity feed because the notification needs to be created.